### PR TITLE
UI View marker

### PR DESCRIPTION
**Pull request by developerark**
Currently you can only add a UIImage as an Annotation marker. I find it very useful to be able to add UIView as an annotation marker since it is more configurable than UIImage.

### CODE REVIEW

File: ARCL/Source/LocationNode.swift [hunk 1]
```diff
@@ -89,8 +89,32 @@ open class LocationAnnotationNode: LocationNode {
...
+    public convenience init(location: CLLocation?, markerView: UIView){
+        self.init(location: location, image: markerView.image() ?? UIImage())
+    }
 
     required public init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 }
+
+extension UIView{
```
Comment (gauravkeshre) [inline ARCL/Source/LocationNode.swift]: As this library supports iOS 11 and above we can conveniently use the newly introduced apis to convert View hierarchy to Image

you can re-write this extension (as I have used in my  version of code)

```
public func renderToImage(afterScreenUpdates: Bool = true) -> UIImage {
        let rendererFormat = UIGraphicsImageRendererFormat.default()
        rendererFormat.opaque = false
        let renderer = UIGraphicsImageRenderer(size: bounds.size, format: rendererFormat)
        
        let snapshotImage = renderer.image { _ in
            drawHierarchy(in: bounds, afterScreenUpdates: afterScreenUpdates)
        }
        return snapshotImage
    }
```

Comment (developerark) [reply ARCL/Source/LocationNode.swift]: I reviewed your code. I see what you are trying to do. Will update my PR with the necessary changes and tests.

**Comment by developerark** [comment]
@gauravkeshre @ProjectDent 
Okay, so I have been playing around trying to add UIView for a few days now. I think the solution is pretty simple and we were all in the weeds trying to take snapshots and the snapshot image every time there is an update in the view. 
The good news is that you can set SCNPlane's firstMaterial.diffuse.contents to a UIView instance. The view updates in the scene as the contents/data model of the view changes. See attached image in which the marker is a custom UIView.
img_1767

**Comment by gauravkeshre** [comment]
Nice @developerark 👍
I will try to replicate this.

This is actually pretty cool as I have lazily loaded images in the view. 
if everything works as expected, I will not need to download and cache the images in advance before rendering on the Anchors.

I will update with my observation in a day or two.

**Comment by developerark** [comment]
@gauravkeshre check PR #150 that has this implementated plus notification on node touch.

**Pull request closed without merging**